### DISCUSSION TITLE
How to compile the "C" source code in Theia web IDE directly

### DISCUSSION THREAD

**Discussion by bluejade008** [eclipse-theia/theia#10485] [Q&A]
Hi specialist,

I have looked through the Theia document, and some issues, discussion, however I have not yet to find out one way which is used to be configured C or C++ comiler in the Theia web IDE, I'm urge to learn the feature, would you like to give some advice?

Thanks.

**Comment by msujew** [comment]
@bluejade008 Just like in vscode, language/compiler support isn't added through the framework, but instead with extensions. For C language support, you could install the [clangd extension](https://open-vsx.org/extension/llvm-vs-code-extensions/vscode-clangd). Regarding compiler support: You need to install the necessary compilers yourself and call them via the terminal. Microsoft has [great documentation on that topic](https://code.visualstudio.com/docs/languages/cpp#_install-a-compiler). One caveat is that you aren't allowed to use the official C/C++ extension made by Microsoft, as their terms of use prohibit usage outside of official VSCode distributions.

**Reply by bluejade008** [reply]
I had installed clangd extension, but I have not yet to compile one example programm named "hellowrold" when I click the menu item [terminal]-->[run build task]. 
Especial one file name "tasks.json" was created when I try to run build task, I opened the tasks.json file and found out a field "problemMatcher": [
                "$eslint-compact"
            ]
I try to modify the "$eslint-compact" into "g++", however the Theia showed me imposible to select g++. 
How can I configure the g++ or gcc in the Theia web IDE?  Whatever g++, gcc and clangd have been intalled the Linux OS.


![clangd](https://user-images.githubusercontent.com/19681325/143828511-0b97964a-3470-487f-80c5-eb53e66bd2ae.JPG)
